src/main/java/com/mraof/minestuck/entity/LotusFlowerEntity.java

```diff
@@ -182,16 +192,27 @@ protected void updateAndSendAnimation(Animation animation)
 		MSPacketHandler.sendToTracking(packet, this);
 	}
 	
+	/**
+	 * Spawns loot from the LOTUS_FLOWER_DEFAULT loot table
+	 */
 	protected void spawnLoot()
 	{
-		Level level = this.level;
-		Vec3 posVec = this.position();
-		
-		ItemEntity unpoweredComputerItemEntity = new ItemEntity(level, posVec.x(), posVec.y() + 1D, posVec.z(), new ItemStack(MSItems.COMPUTER_PARTS.get(), 1));
-		level.addFreshEntity(unpoweredComputerItemEntity);
-		
-		ItemEntity sburbCodeItemEntity = new ItemEntity(level, posVec.x(), posVec.y() + 1D, posVec.z(), new ItemStack(MSItems.SBURB_CODE.get(), 1));
-		level.addFreshEntity(sburbCodeItemEntity);
+		if(!level.isClientSide)
+		{
+			ServerLevel serverWorld = (ServerLevel) level;
+			
+			LootTable lootTable = serverWorld.getServer().getLootTables().get(MSLootTables.LOTUS_FLOWER_DEFAULT);
+			List<ItemStack> loot = lootTable.getRandomItems(new LootContext.Builder(serverWorld).create(LootContextParamSets.EMPTY));
+			if(loot.isEmpty())
+				LOGGER.warn("Tried to generate loot for Lotus Flower, but no items were generated!");
+			
+			for(ItemStack itemStack : loot)
+			{
+				if(itemStack.getItem() == MSItems.SBURB_CODE.get())
+					SburbCodeItem.setParadoxInfo(itemStack, true);
```

Comment:
This seems better suited for a loot function

Reply:
Now using a function